docs/source/concept_guides/deferring_execution.md

```diff
@@ -108,3 +108,19 @@ with accelerator.main_process_first():
         remove_columns=["idx", "sentence1", "sentence2"],
     )
 ```
+
+## Applying checks such as Early Stopping
+
+To have a check that works with a flag set by a particular process, the `check` and `set` breakpoint API should be used.
```

Comment:
IMO, as a reader, I would be a bit confused about when I have to use this and when not. For instance, IIUC, it is not necessarily true that early stopping requires this feature. E.g. if the user collects some metrics and correctly applies `gather_for_metrics`, all processes should be synced up and there is no danger of getting a stopping signal on one process but not the others. IIUC, the initial issue the user had was caused by directly checking the loss, which can differ between processes.

Reply:
Expanded on this a bit more